dojo/tools/vulners/parser.py

```diff
@@ -0,0 +1,99 @@
+import json
+import logging
+from cvss.cvss3 import CVSS3
+
+from dojo.models import Endpoint, Finding
+from dojo.tools.vulners.importer import VulnersImporter
+
+logger = logging.getLogger(__name__)
+
+
+vulners_severity_mapping = {
+    1: 'Info',
+    2: 'Low',
+    3: 'Medium',
+    4: 'High',
+    5: 'Critical'
+}
+
+
+class VulnersParser(object):
+    """Parser that can load data from Vulners Scanner API"""
+
+    def get_scan_types(self):
+        return ["Vulners"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "Vulners"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import Vulners Audit reports in JSON."
+
+    def requires_tool_type(self, scan_type):
+        return "Vulners"
+
+    def requires_file(self, scan_type):
+        return False
+
+    def get_findings(self, file, test):
+        # API export is a JSON file
+        if file:
+            data = json.load(file)
+        else:
+            data = VulnersImporter().get_findings(test)
+
+        findings = []
+        vulns = {}
+        report = data.get("data", dict()).get("report", list())
+
+        if not file:
+            vulns_id = [vuln.get("vulnID") for vuln in report]
+            vulns = VulnersImporter().get_vulns_description(test, vulns_id).get('data', dict()).get('documents', dict())
+
+        # for each issue found
+        for component in report:
+            id = component.get("vulnID")
+            vuln = vulns.get(id, dict())
+            title = component.get("title", id)
+            family = component.get("family")
+            agentip = component.get("agentip")
+            agentfqdn = component.get("agentfqdn")
+            severity = vulners_severity_mapping[component.get("severity", 0)]
+
+            finding = Finding(
+                title=title,
+                severity=severity,
+                impact=severity,
+                description=vuln.get("description", title),
+                mitigation=component.get("cumulativeFix"),
+                static_finding=False,  # by definition
+                dynamic_finding=True,  # by definition
+                # false_p=False,
+                # duplicate=False,
+                out_of_scope=False,
+                vuln_id_from_tool=id,
```

Comment:
it seems that it should be better to use `unique_id_from_tool` for this data.

Reply:
I kept **vulnerability_id_from_tool** and added vulners.com to **VULNERABILITY_URLS**, because I believe it's exactly public vulnerability ID like [RHSA-2018:2285](https://access.redhat.com/errata/RHSA-2018:2285) or [CESA-2021:0348](https://lists.centos.org/pipermail/centos-announce/2021-February/060805.html) and not just internal identifier

Reply:
ok. thanks